pkg/volume/util/subpath/subpath_windows.go

```diff
@@ -342,6 +348,10 @@ func doSafeMakeDir(pathname string, base string, perm os.FileMode) error {
 		if stat.Mode()&os.ModeSymlink != 0 {
 			return fmt.Errorf("subpath %q is an unexpected symlink after Mkdir", currentPath)
 		}
+		// go1.23 behavior change: https://github.com/golang/go/issues/63703#issuecomment-2535941458
+		if stat.Mode()&os.ModeIrregular != 0 {
```

Comment:
Do we need to change this file? https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/mounttest/mt_utils_windows.go#L80

Reply:
in the end, yes, and that's only for test, so we could change it later